### PR TITLE
Revise TotalCircSupply

### PR DESCRIPTION
**Pull request by arajasek**
TotalCircSupply is still calculated as `filVested + filMined - filBurnt - filLocked`, but these values are now obtained as follows:

- `filVested = multisigVested + accountVested + genesisLocked`, where `multisigVested` is the total vested funds of the genesis multisigs at a given epoch, `accountVested` is the sum over all **account** actors in the genesis state of `max(currBalance - genesisBalance, 0)`, and `genesisLocked` is the locked funds in the power and market actor at genesis.
- `filMined` is now obtained from the Reward Actor state (https://github.com/filecoin-project/lotus/pull/2605#discussion_r461976438).
- `filBurnt` is (still) just the balance of the Burnt Funds actor. We may want to change it to be the Burnt Funds currBalance - genesisBalance if there's a possibility that it's genesis balance could be non-zero.
- `filLocked` is the locked funds in the power and market states at the given height.

We might wanna update the CE supply doc to spell this out.

### CODE REVIEW

File: chain/stmgr/stmgr.go [hunk 1]
```diff
@@ -829,40 +858,185 @@ func (sm *StateManager) setupGenesisMsigs(ctx context.Context) error {
...
+// - For Multisigs, it counts the actual amounts that have vested at the given epoch
+// - For Accounts, it counts max(currentBalance - genesisBalance, 0).
+func (sm *StateManager) GetFilVested(ctx context.Context, height abi.ChainEpoch, st *state.StateTree) (abi.TokenAmount, error) {
+	vf := big.Zero()
+	for _, v := range sm.genInfo.genesisMsigs {
+		au := big.Sub(v.InitialBalance, v.AmountLocked(height))
+		vf = big.Add(vf, au)
+	}
+
+	for _, v := range sm.genInfo.genesisActors {
```
[the remark this answers is not in the document]
Comment (arajasek) [reply chain/stmgr/stmgr.go:909]: Strong agree.

File: chain/stmgr/stmgr.go [hunk 2]
```diff
@@ -820,40 +849,179 @@ func (sm *StateManager) setupGenesisMsigs(ctx context.Context) error {
...
+		}
+
+		// TODO: We should probably slowly phase initBal into TotalCircSupply, instead of considering them locked forever
+		diff := big.Sub(v.initBal, act.Balance)
+		if diff.GreaterThan(big.Zero()) {
+			vf = big.Add(vf, diff)
+		}
+	}
+
+	vf = big.Add(vf, sm.genInfo.genesisPledge)
```
Comment (zixuanzh) [inline chain/stmgr/stmgr.go:915]: maybe the above TODO are for these lines but for fil locked, they will enter circulating supply as sectors expire.

Comment (arajasek) [reply chain/stmgr/stmgr.go:914]: sorry, that should've been removed

**Comment by zixuanzh** [comment]
This is already doing what you are suggesting as lockup reward is deducted through FIL_Locked

**Comment by zixuanzh** [comment]
This is already doing what you are suggesting as vesting reward is deducted through FIL_Locked

**Pull request merged**